server.go

```diff
@@ -936,3 +756,280 @@ func (s *Server) handlerHistory(w http.ResponseWriter, r *http.Request) {
+
+	if !jikanEnabled {
+		s.l.Debug("Aborting Jikan autofill since it is not enabled")
+		data.ErrorMessage = append(data.ErrorMessage, "Jikan API usage was not enabled by the site administrator")
+		return nil, fmt.Errorf("Jikan not enabled")
+	} else {
+		// Get Data from MAL (jikan api)
+		rgx := regexp.MustCompile(`[htp]{4}s?:\/\/[^\/]*\/anime\/([0-9]*)`)
+		match := rgx.FindStringSubmatch(sourcelink)
+		var id string
+		if len(match) < 2 {
+			s.l.Debug("Regex match didn't find the anime id in %v", sourcelink)
+			data.ErrorMessage = append(data.ErrorMessage, "Could not retrive anime id from provided link, did you input a manga link?")
+			data.ErrLinks = true
+			return nil, fmt.Errorf("Could not retrive anime id from link")
+		} else {
+			id = match[1]
+		}
+
+		bannedTypesString, err := s.data.GetCfgString(ConfigJikanBannedTypes, DefaultJikanBannedTypes)
+
+		if err != nil {
+			s.l.Error("Error while retriving config value 'JikanBannedTypes':\n %v", err)
+			s.doError(
+				http.StatusInternalServerError,
+				"something went wrong :C",
+				w, r)
+			return nil, err
+		}
+
+		bannedTypes := strings.Split(bannedTypesString, ",")
+
+		maxEpisodes, err := s.data.GetCfgInt(ConfigJikanMaxEpisodes, DefaultJikanMaxEpisodes)
+
+		if err != nil {
+			s.l.Error("Error while retriving config value 'JikanMaxEpisodes':\n %v", err)
+			s.doError(
+				http.StatusInternalServerError,
+				"something went wrong :C",
+				w, r)
+			return nil, err
+		}
+
+		sourceAPI := jikan{id: id, l: s.l, excludedTypes: bannedTypes, maxEpisodes: maxEpisodes}
+
+		// Request data from API
+		results, err := getMovieData(&sourceAPI)
+
+		if err != nil {
+			s.l.Error("Error while accessing Jikan API: %v", err)
+			data.ErrorMessage = append(data.ErrorMessage, err.Error())
+			return nil, err
+		}
+
+		return results, nil
+	}
+}
+
+func (s *Server) handleTmdb(data *dataAddMovie, w http.ResponseWriter, r *http.Request, sourcelink string) ([]string, error) {
+
+	tmdbEnabled, err := s.data.GetCfgBool("TmdbEnabled", DefaultTmdbEnabled)
+	if err != nil {
+		s.l.Error("Error while retriving config value 'TmdbEnabled':\n %v", err)
+		data.ErrorMessage = append(data.ErrorMessage, "Something went wrong :C")
+		return nil, err
+	}
+
+	if !tmdbEnabled {
+		s.l.Debug("Aborting Tmdb autofill since it is not enabled")
+		data.ErrorMessage = append(data.ErrorMessage, "Tmdb API usage was not enabled by the site administrator")
+		return nil, fmt.Errorf("Tmdb not enabled")
+	} else {
+		// Retrieve token from database
+		token, err := s.data.GetCfgString("TmdbToken", "")
+		if err != nil || token == "" {
+
+			s.l.Debug("Aborting Tmdb autofill since no token was found")
+			data.ErrorMessage = append(data.ErrorMessage, "TmdbToken is either empty or not set in the admin config")
+			return nil, fmt.Errorf("TmdbToken is either empty or not set in the admin config")
+		}
+		// get the movie id
+		rgx := regexp.MustCompile(`[htp]{4}s?:\/\/[^\/]*\/title\/(tt[0-9]*)`)
```

Comment:
Another regex that shouldn't be built on every run.

Reply:
Will be fixed in the next commit